### PR TITLE
KAFKA-15215: docs for KIP-954

### PR DESCRIPTION
**Pull request by agavra**

### CODE REVIEW

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -310,6 +310,19 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
...
           </tr>
+          <tr class="row-even"><td>default.dsl.store</td>
+            <td>Low</td>
+            <td colspan="2">(Deprecated) The default state store type used by DSL operators.</td>
+            <td><code>ROCKS_DB</code></td>
+          </tr>
+          <tr class="row-odd"><td>dsl.store.suppliers.class</td>
+            <td>Low</td>
+            <td colspan="2">
+              Defines which store implementations to plug in to DSL operators. Must implement
```
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
              Defines a default state store implementation to be used by any stateful DSL operator
              that has not explicitly configured the store implementation type. Must implement
```

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -310,6 +310,19 @@ <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="header
             <td colspan="2">Default serializer/deserializer for the inner class of windowed values, implementing the <code class="docutils literal"><span class="pre">Serde</span></code> interface.</td>
             <td>null</td>
           </tr>
+          <tr class="row-even"><td>default.dsl.store</td>
+            <td>Low</td>
+            <td colspan="2">(Deprecated) The default state store type used by DSL operators.</td>
```
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
            <td colspan="2">
              [DEPRECATED] The default state store type used by DSL operators. Deprecated in
              favor of <code>dsl.store.suppliers.class</code>
              </td>
```

File: docs/streams/upgrade-guide.html [hunk 3]
```diff
@@ -185,6 +185,13 @@ <h3><a id="streams_api_changes_370" href="#streams_api_changes_370">Streams API
     </pre>
     </p>
 
+    <p>
+        <code>default.dsl.store</code> was replaced with <code>dsl.store.suppliers.class</code> to allow for custom store configuration.
```
Comment (ableegoldman) [inline docs/streams/upgrade-guide.html]:
```suggestion
        The <code>default.dsl.store</code> config was deprecated in favor of the new
        <code>dsl.store.suppliers.class</code> config to allow for custom state store 
        implementations to be configured as the default.
        
```

**Review by ableegoldman — COMMENTED** [review]
Just some minor wording suggestions to make the deprecation stuff clear, otherwise LGTM

**Comment by mjsax** [comment]
@agavra -- PR need to be rebased.

**Comment by agavra** [comment]
@mjsax rebased - note that `trunk` had a "bug" where the latest PR added:
```
<h3><a id="streams_api_changes_370" href="#streams_api_changes_370">Streams API changes in 3.7.0</a></h3>
```
even though it already existed, so I removed that line.

**Comment by ableegoldman** [comment]
Ok this has dragged on for long enough, I'm just going to merge this even though we haven't gotten a single individual run without any build failures, because (a) we have at least one yellow build without compiler errors for each JDK if you look across the last three runs, and (b) this is only touching docs anyway so the failures are pretty clearly unrelated

cc @stanislavkozlovski going to merge this to 3.7 since it's just feature docs for 3.7

**Comment by ableegoldman** [comment]
FYI @agavra try to always fill out a PR description, even if it's super short/simple, so we have something for the commit message. I'll just make it up for this case but going forward it's on you 😛

**Comment by ableegoldman** [comment]
Merged to trunk and cherrypicked to 3.7

**Pull request merged**